crates/protocol/src/block.rs

```diff
@@ -53,12 +55,39 @@ pub struct L2BlockInfo {
     /// The L1 origin [BlockID]
     pub l1_origin: BlockID,
     /// The sequence number of the L2 block
-    pub seq_num: U64,
+    #[cfg_attr(feature = "serde", serde(deserialize_with = "deserialize_u64_hex"))]
+    pub seq_num: u64,
```

Comment:
I think all of these can be replaced with
```
    #[serde(with = "alloy_serde::quantity")]
```

Reply:
Amazing thank you